### PR TITLE
Process finalized checkpoints in order

### LINKED ISSUE #2015: Error processing finalized checkpoint
I got this when syncing pyrmont locally:

```
2021-01-31 20:36:17 [SYNC]             info: Sync progress 1957/16662 - 1.1 days left - 5.08 slots/s
2021-01-31 20:36:18 [CHORES]          error: Error processing finalized checkpoint epoch=1953 code=PROTO_ARRAY_ERROR_FINALIZED_NODE_UNKNOWN, root=0x631413d98355f04a73d701e8c19c3428fc1ef6702f032dc125883fea65f1e814
Error: PROTO_ARRAY_ERROR_FINALIZED_NODE_UNKNOWN
    at ProtoArray.maybePrune (/Users/tuyennguyen/Projects/workshop/lodestar/packages/lodestar-fork-choice/src/protoArray/protoArray.ts:241:13)
    at LodestarForkChoice.prune (/Users/tuyennguyen/Projects/workshop/lodestar/packages/lodestar-fork-choice/src/forkChoice/forkChoice.ts:488:28)
    at ChainEventEmitter.<anonymous> (/Users/tuyennguyen/Projects/workshop/lodestar/packages/lodestar/src/tasks/index.ts:108:29)
2021-01-31 20:36:32 [CHAIN]            info: Checkpoint justified epoch=1959, root=0x85a38819f760677973679cb24892959e521b46949a6acd54957cc5ce14636d1b
```

### PR DESCRIPTION
**Pull request by twoeths**
resolves #2015 
+ use `JobQueue` to order the finalized checkpoint execution: one finalized checkpoint needs to be processed completely before the next one

### CODE REVIEW

**Comment by dapplion** [comment]
@wemeetagain Can you review? I keep seeing this error, would be great to fix it

**Review by dapplion — DISMISSED** [review]

**Comment by wemeetagain** [comment]
LGTM minus file conflict

File: packages/lodestar/src/tasks/index.ts [hunk 1]
```diff
@@ -13,6 +14,12 @@ import {StatesArchiver} from "./tasks/archiveStates";
 import {IBeaconSync} from "../sync";
 import {InteropSubnetsJoiningTask} from "./tasks/interopSubnetsJoiningTask";
 import {INetwork, NetworkEvent} from "../network";
+import {JobQueue} from "../util/queue";
+
+/**
+ * Minimum number of epochs between archived states
+ */
+export const MIN_EPOCHS_PER_DB_STATE = 1024;
```
Comment (3xtr4t3rr3str14l) [inline packages/lodestar/src/tasks/index.ts]: this is the same value and concept as `PERSIST_STATE_EVERY_EPOCHS` no?  why not just use that const instead?

**Review by dapplion — DISMISSED** [review]

File: packages/lodestar/src/tasks/index.ts (hunk 1 above)
Comment (wemeetagain) [reply packages/lodestar/src/tasks/index.ts]: I think these lines can be deleted?

**Review by dapplion — APPROVED** [review]

**Review by wemeetagain — APPROVED** [review]

**Pull request merged**